File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.java
class Solution {
    public int findMin(int[] nums) {
        int h=nums.length-1;
        int l=0;
        int mid=0;
        int ans=Integer.MAX_VALUE;
        while(h>=l)
        {
            mid = (h+l)/2;
            int m = nums[mid];
            if(nums[mid]<nums[h])
            {
                h = mid-1;
            }
            else
            {
                l = mid+1;
            }
            ans = Math.min(ans,m);
        }
        return ans;
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-2.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        maxval = nums[0]
        minval = nums[0]
        ans = nums[0]
        for i in range(1,len(nums)):
            tmp_min = minval
            tmp_max = maxval
            maxval = max(tmp_max*nums[i],max(tmp_min*nums[i],nums[i]))
            minval = min(tmp_min*nums[i],min(tmp_max*nums[i],nums[i]))
            ans = max(ans,maxval)
            
        return ans